bittensor/core/async_subtensor.py

```diff
@@ -279,20 +405,41 @@ async def get_hyperparameter(
         block_hash: Optional[str] = None,
         reuse_block: bool = False,
     ) -> Optional[Any]:
-        """
-        Retrieves a specified hyperparameter for a specific subnet.
+        """Retrieves a specified hyperparameter for a specific subnet.
 
-        Arguments:
-            param_name (str): The name of the hyperparameter to retrieve.
+        This method queries the blockchain for subnet-specific hyperparameters such as difficulty, tempo, immunity
+        period, and other network configuration values.
+
+        Args:
+            param_name (str): The name of the hyperparameter to retrieve (e.g., "Difficulty", "Tempo",
+                "ImmunityPeriod").
             netuid (int): The unique identifier of the subnet.
-            block: the block number at which to retrieve the hyperparameter. Do not specify if using block_hash or
-                reuse_block
-            block_hash (Optional[str]): The hash of blockchain block number for the query. Do not specify if using
-                block or reuse_block
-            reuse_block (bool): Whether to reuse the last-used block hash. Do not set if using block_hash or block.
+            block (Optional[int]): The block number at which to retrieve the hyperparameter. Do not specify if using
```

Comment:
```suggestion
            block: The block number at which to retrieve the hyperparameter. Do not specify if using
```
we want to remove all annotations from docstrings. Please apply this approach for whole PR